app/code/Magento/InventoryCatalogSearch/Plugin/Search/FilterMapper/AdaptStockStatusFilterPlugin.php

```diff
@@ -77,7 +88,9 @@ public function __construct(
      * @param string $type
      * @param bool $showOutOfStockFlag
      * @return Select
-     * @throws \InvalidArgumentException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     * @throws \Zend_Db_Select_Exception
```

Comment:
we should not throw Zend_Db_Select_Exception (as any other Zend exception) from any class, such low type exception should be caught
 and wrapped with a higher level Exception